shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -786,29 +786,17 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
 void FlutterPlatformViewsController::BringLayersIntoView(LayersMap layer_map) {
   FML_DCHECK(flutter_view_);
   UIView* flutter_view = flutter_view_.get();
-  auto zIndex = 0;
   // Clear the `active_composition_order_`, which will be populated down below.
   active_composition_order_.clear();
   for (size_t i = 0; i < composition_order_.size(); i++) {
     int64_t platform_view_id = composition_order_[i];
     std::vector<std::shared_ptr<FlutterPlatformViewLayer>> layers = layer_map[platform_view_id];
     UIView* platform_view_root = root_views_[platform_view_id].get();
-
-    if (platform_view_root.superview != flutter_view) {
-      [flutter_view addSubview:platform_view_root];
-    }
-    // Make sure the platform_view_root is higher than the last platform_view_root in
-    // composition_order_.
-    platform_view_root.layer.zPosition = zIndex++;
-
+    // Simply having a higher zPosition is not enough since it does not affect touch behavior.
+    // `addSubview` will automatically reorder subview if it is already added.
+    [flutter_view addSubview:platform_view_root];
     for (const std::shared_ptr<FlutterPlatformViewLayer>& layer : layers) {
-      if ([layer->overlay_view_wrapper.get() superview] != flutter_view) {
-        [flutter_view addSubview:layer->overlay_view_wrapper];
-      }
-      // Make sure all the overlays are higher than the platform view.
-      layer->overlay_view_wrapper.get().layer.zPosition = zIndex++;
-      FML_DCHECK(layer->overlay_view_wrapper.get().layer.zPosition >
-                 platform_view_root.layer.zPosition);
+      [flutter_view addSubview:layer->overlay_view_wrapper];
```

Comment:
Can you manually test that this change didn't regress https://github.com/flutter/flutter/issues/86787 (see #29930)

Reply:
I can confirm that it didn't regress https://github.com/flutter/flutter/issues/86787. 

Though I found a different regression (see pic below). The navigation bar becomes transparent, so the platform view is displayed underneath. This seems to be similar to https://github.com/flutter/flutter/issues/119485. 

It is NOT caused by this PR because I get the same behaviors with or without this change; but it is a regression, because the released build is working fine. I will comment in that issue about this finding. 

![Simulator Screen Shot - iPhone 14 Pro - 2023-02-21 at 13 32 53](https://user-images.githubusercontent.com/41930132/220464012-e1742968-7dfc-4e4a-9ce6-2b3261be7ab6.png)